llvm/include/llvm/Support/BlockFrequency.h

```diff
@@ -93,6 +94,28 @@ class BlockFrequency {
     return *this;
   }
 
+  /// Returns true if `this` is greater than `other` and the magnitude of the
+  /// difference falls within the topmost `SignificantBits` of `this`.
+  ///
+  /// The formula is related to comparing a "relative change" to a threshold.
+  /// When choosing the threshold as a negative power-of-two things can be
+  /// computed cheaply:
+  ///   with A = this->Frequency; B = Other.Frequency; T = SignificantBits
+  ///   relative_change(A, B) = abs(A - B) / A
```

Comment:
I think we can make this reflexive by changing the formular to 
`relative_change(A, B) = abs(A - B) / min(A, B)`. This should simplify your code below too.

Reply:
Well I thought that saving the `if` may be worth it for some callers, but you are right that it happened to produce extra complications with that `IsEHPad` logic here. I'll change it.